src/precice/impl/WriteDataContext.hpp

```diff
@@ -32,6 +32,22 @@ class WriteDataContext : public DataContext {
    */
   mesh::PtrData providedData();
 
+  /**
+   * @brief Store values in _providedData.values()
+   *
+   * @param[in] indices ids of data
+   * @param[in] values values of data
+   */
+  void writeValues(std::vector<int> indices, const Eigen::Map<const Eigen::VectorXd> values);
+
+  /**
+   * @brief Store gradients in _providedData.gradientValues()
+   *
+   * @param[in] indices ids of data
+   * @param[in] gradients gradients of data
+   */
+  void writeGradientValues(std::vector<int> indices, const Eigen::Map<const Eigen::MatrixXd> gradients);
```

Comment:
These are potentially large data sets. I would rather use a reference instead of copies?!

Reply:
As far as I understand an `Eigen::Map` works more or less like a reference:

> While one option is to make a copy of the data, most commonly you probably want to re-use this memory as an Eigen type. Fortunately, this is very easy with the [Map](https://eigen.tuxfamily.org/dox/classEigen_1_1Map.html) class.

https://eigen.tuxfamily.org/dox/group__TutorialMapClass.html

Reply:
Done. [5cec86c](https://github.com/precice/precice/pull/1633/commits/5cec86c584d550de4f13ea538308819dd343a3ee)